### ISSUE TITLE
[camera_windows] Failed to initialize camera: camera_error: Failed to create camera

### ISSUE THREAD

**Issue by LWY6670** [flutter/flutter#103897]
When I ran the example in [packages\camera\camera_windows\]:

**[Create camera]** do NOT work for **_1_ of my three** cameras.
You CAN [Switch to] it, but if you [Create camera], it says:

>Failed to initialize camera: camera_error: Failed to create camera.

(No specific ERROR info). It is a USB UVC camera.

The other 2 cameras(1 built in, 1 USB) work well.
The camera with error works well in other programs such as OBS Studio and some C++/C# camera video capture examples.

![image](https://user-images.githubusercontent.com/27002679/168595587-c6cb9772-6e7a-4f9d-a7af-741a9dc8d62a.png)

**Comment by maheshmnj** [comment]
Hi @LWY6670, Thanks for filing the issue. Does the camera fail to initialize the primary camera or the external camera? Please share the output of `flutter doctor -v` along with the output of `flutter run -v` in a text file.

**Comment by LWY6670** [comment]
Thank you @maheshmnj for all. 

That camera is an external camera. In fact, it is a video capture (It looks like a USB thumb drive). I know many people use them to do their webcast. 

In order to confirm this issue, I repeated the test using another Video Capture Card (I inserted it into the PCI-e slot inside another computer, a desktop), the same thing happened.
[Message produced by Flutter.txt](https://github.com/flutter/flutter/files/8704661/Message.produced.by.Flutter.txt)
![image](https://user-images.githubusercontent.com/27002679/168716829-059ee572-5329-4060-bf6c-a09f73ec0b61.png)
![image](https://user-images.githubusercontent.com/27002679/168716862-3a3da2d4-59ce-4281-8a5c-a9ca2000ba68.png)
![image](https://user-images.githubusercontent.com/27002679/168716880-63756c78-84d7-4838-86d0-fac05b0b7b4d.png)

**Comment by maheshmnj** [comment]
I see, thanks for the info @LWY6670, Can you please check if the external camera works in other apps? I am wondering if this is in any way related to camera permissions?

**Comment by LWY6670** [comment]
Yes. The external camera works in other apps, NO ERROR, along with the other 2 cameras(1 built-in, 1 external) in the computer。Sometime, Windows asked me to confirm to open a camera, I answered Yes.

I tested the cameras on two computers, using at least 5 different other programs, below are 2 of them:
(1) OBS: from https://obsproject.com/ 
(2) C# sample, from https://github.com/simongh/DirectShow.Capture/tree/master/Sample 

Here is the snapshot of the C# app, using THE error-with-flutter camera, it is good-with-c#-and-other-all:
![image](https://user-images.githubusercontent.com/27002679/169019385-20383c7a-1a1b-43d1-8508-6575caba139f.png)

All cameras works smoothly with all other programs I tested. No exception.

I think maybe you overlooked a very **IMPORTANT** detail in my response:
The error camera is NOT a normal external camera. It is a VIDEO CAPTURE CARD.
It is in compliance with USB Video Class (**UVC**) and **DirectShow** specification or standard. 
It is a camera indeed, so camera_windows can find it and add it to the camera list.

Every normal camera works well with camera_windows, 
But, both of the UVC Cameras (one USB slot, one PCIe slot) have ERROR with camera_windows,
Another clue is that, in many case, a VIDEO CAPTURE CARD will be more expensive than a normal camera.

The C# (open source on GitHub: https://github.com/simongh/DirectShow.Capture/ ) sample do the same thing using DirectShow, so there is no error with it.

So, I believe, the **DirectShow** is the key factor to fix it.

**Comment by maheshmnj** [comment]
Thanks for the info @LWY6670, Labeling this issue for further investigation from the team.

**Comment by LWY6670** [comment]
Thank you @maheshmnj , here is some further information.

The captions appeared in the apps for the cameras are: 
- [**_USB Video_**] for the USB-slot camera 
- [DC1150 USB, Analog Capture]) for the PCIe slot camera

I tested them again using a **_C++ open source example_** from  

- https://github.com/wenyec/UVCViwer   // which uses **_DirectShow_** too
- whose README.md says: The UVC Viewer for the USB camera **especially USB3.** It derived from the **MC sample AMCap (DirectShow).**

I compiled it with the **Windows-x64** platform config using VS 2019, the cameras work normally as well.

![image](https://user-images.githubusercontent.com/27002679/169063333-520da685-d0b5-4c02-8a40-54e15854a58f.png)

![image](https://user-images.githubusercontent.com/27002679/169063409-475f03a2-20b4-4dae-9057-8ffaa0165fbc.png)

**Comment by zhhding** [comment]
HI,  @LWY6670 兄弟,这个问题解决了没,我这边遇到同样的问题,摄像头无法初始化.请问您那边有解决方案了没,

**Comment by LWY6670** [comment]
@zhhding  还没有。在等 Flutter 包的升级。

**Comment by yiky84119** [comment]
![image](https://github.com/flutter/flutter/assets/3356845/07d45481-737a-488d-8c8e-ed1487330dd0)

fail to GetAvailableDeviceMediaType

**Comment by On365** [comment]
Any news on this?
I want to use a flutter app with a video capture card
